libs/framework/include/celix/Properties.h

```diff
@@ -253,6 +259,38 @@ namespace celix {
         std::size_t size() const {
             return celix_properties_size(cProps.get());
         }
+
+        /**
+         * @brief Converts the properties a (new) std::string, std::string map.
+         */
+        std::map<std::string, std::string> convertToMap() const {
+            std::map<std::string, std::string> result{};
+            for (const auto& pair : *this) {
+                result[pair.first] = pair.second;
+            }
+            return result;
+        }
+
+        /**
+         * @brief Converts the properties a (new) std::string, std::string unordered map.
+         */
+        std::unordered_map<std::string, std::string> convertToUnorderedMap() const {
+            std::unordered_map<std::string, std::string> result{};
+            for (const auto& pair : *this) {
+                result[pair.first] = pair.second;
+            }
+            return result;
+        }
+
+        /**
+         * @brief cast the celix::Properties to a std::string, std::string map.
+         * @warning This method is added to ensure backwards compatibility with the celix::dm::Properties, but the
+         * use of this cast should be avoided.
+         * This method will eventually be removed.
+         */
+        operator std::map<std::string, std::string>() const {
```

Comment:
Without the `explicit` keyword, this results in interesting behaviour where it might not be expected. While providing a good backwards-compatible approach, we might want to think about some ifdef construction where the explicit keyword is added if enabled. This will allow easier migration in the future.

Reply:
Could do it in this PR, could also make a new ticket like #327

Reply:
Agree. I added a define for this.

We can enable this - for now - downstream and eventually remove this.